pkg/kv/kvserver/replica_proposal_buf.go

```diff
@@ -1175,6 +1178,20 @@ func (rp *replicaProposer) closedTimestampTarget() hlc.Timestamp {
 	return (*Replica)(rp).closedTimestampTargetRLocked()
 }
 
+// maybeRegister is called when a proposal is about to be proposed. It goes
+// through all the entries in the proposal and registers them for tracking if
+// the context was set up for tracking when it was submitted.
+func (rp *replicaProposer) maybeRegister(ctx context.Context, ents []raftpb.Entry) {
+	r := (*Replica)(rp)
+	for _, e := range ents {
+		if id, valid, traced := getID(e); valid && traced {
+			if prop, ok := r.mu.proposals[id]; ok {
+				rp.mu.raftTracer.Register(prop, ents)
```

Comment:
Why `Register` all the entries? We've just figured out that the particular entry `e` is traced, so can register just this one? Otherwise, this code can end up with a quadratic complexity.

Reply:
Fixed. I had change this to only find the correct entry but hadn't updated this code.